src/Common/src/Microsoft/Win32/SafeHandles/SafeFileHandle.Unix.cs

```diff
@@ -35,7 +40,7 @@ internal static SafeFileHandle Open(string path, Interop.Sys.OpenFlags flags, in
             bool enoentDueToDirectory = (flags & Interop.Sys.OpenFlags.O_CREAT) != 0;
 
             // Open the file. 
-            int fd;
+            IntPtr fd;
```

Comment:
Another option we have here is to move the EINTR checks into the shim (obviously with a different change, not this one).  And then managed code shouldn't need to worry about them at all.  Then I think the managed code could be really clean: Open returns a SafeFileHandle and managed wouldn't even need the while loop.  Just a check to throw on an Invalid handle.